course04/homework00/ciaranfrain/sorted_array.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <deque>
+
+class SortedArray
+{
+public:
+	SortedArray(std::size_t maximumSize);
+	
+	void clear() { mStorage.clear(); }
+	const int& operator[](int n) const { return Get(n); }
+
+	bool empty() const { return mStorage.empty(); }
+	std::size_t size() const { return mStorage.size(); }
+
+	void Insert(int n);
+	const int& Get(int index) const;
```

Comment:
Why having `Get` and `operato[]` is they implement the exact same operation? It's an option, but in general at Optiver we stick to the minimum: every public method will have to be unit tested - the less code, the better.